### PR TITLE
fix(bake): add support for full layer path

### PR DESCRIPTION
**Pull request by Sonderwoods**
this allows for sublayers to be created.

tested and should be ok. Not so pythonic =) 

@chriswmackey

### CODE REVIEW

File: ladybug_rhino/bakegeometry.py [hunk 1]
```diff
@@ -137,11 +137,25 @@ def _get_attributes(layer_name=None, attributes=None):
...
+        for i in range(len(part_names)):
+            combined_part_names.append("::".join(part_names[0:i+1])) #  [0] = layer ,  [1] = layer::sublayer, [2] = layer::sublayer::sublayer
+            checking_index = layer_table.FindByFullPath(combined_part_names[i], True)
+            if  checking_index < 0:  # layer::sublayer doesnt exist:
+
+                parent_layer = docobj.Layer()
+                parent_layer.Name = part_names[i]
+                
+                if i > 0:
+                    parent_layer.ParentLayerId = prev_layer.Id # nesting it to previosly created layer
```
Comment (chriswmackey) [inline ladybug_rhino/bakegeometry.py:155]: Should be `previously`

File: ladybug_rhino/bakegeometry.py [hunk 2]
```diff
@@ -137,11 +137,25 @@ def _get_attributes(layer_name=None, attributes=None):
...
-    layer_index = layer_table.Find(layer_name, True)
+    layer_index = layer_table.FindByFullPath(layer_name, True)
+    combined_part_names = []
     if layer_index < 0:
-        parent_layer = docobj.Layer()
-        parent_layer.Name = layer_name
-        layer_index = layer_table.Add(parent_layer)
+        part_names = layer_name.split("::") #  full name could be layer::sublayer::sublayer
+        for i in range(len(part_names)):
+            combined_part_names.append("::".join(part_names[0:i+1])) #  [0] = layer ,  [1] = layer::sublayer, [2] = layer::sublayer::sublayer
```
Comment (chriswmackey) [inline ladybug_rhino/bakegeometry.py:147]: Can ew make sure that all of the lines of code here obey [our hard character limit of 89 characters](https://github.com/ladybug-tools/contributing/wiki/python-style-guide#maximum-line-length)?

**Review by chriswmackey — COMMENTED** [review]
Thanks, @Sonderwoods , and sorry for the late response.

Your code generally looks good and we can merge it as soon as you fix those two styling issues that I commented on there.

**Comment by Sonderwoods** [comment]
Fixed

**Comment by Sonderwoods** [comment]
@chriswmackey  for some reason i can't bake hatches anymore in newest version. Goes for bake legend and the mesh-to-hatch component. I suspect this "fix" has ruined that? I'm unsure, also unsure if I'm the only one with the issue.

**Pull request closed without merging**